src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb

```diff
@@ -3852,10 +3852,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
                 Dim fullDecl = Me.GetFullDeclaration(declaration)
 
                 ' special handling for replacing at location of a sub-declaration
-                If fullDecl IsNot declaration Then
+                If fullDecl IsNot declaration AndAlso fullDecl.IsKind(newFullDecl.Kind) Then
 
                     ' try to replace inline if possible
-                    If fullDecl.IsKind(newFullDecl.Kind) AndAlso GetDeclarationCount(newFullDecl) = 1 Then
+                    If GetDeclarationCount(newFullDecl) = 1 Then
```

Comment:
same question.

Reply:
Unit test added to SyntaxGeneratorTests.vb for the failing scenario.

---
In reply to: [230607049](https://github.com/dotnet/roslyn/pull/30777#discussion_r230607049) [](ancestors = 230607049)